File: Week 2/4_least_common_multiple/lcm.py
# Uses python3
import sys

def lcm_naive(a, b):
    for l in range(1, a*b + 1):
        if l % a == 0 and l % b == 0:
            return l

    return a*b


def gcd(a, b):
    if not b:
        return a
    return gcd(b, a%b)


def lcm_fast(a, b):
    h = gcd(a, b)
    return (a * (b // h))


if __name__ == '__main__':
    # input = sys.stdin.read()
    a, b = map(int, input().split())
    print(lcm_fast(a, b))

File: Week 2/4_least_common_multiple/lcm.cpp
#include <iostream>

long long lcm_naive(int a, int b) {
  for (long l = 1; l <= (long long) a * b; ++l)
    if (l % a == 0 && l % b == 0)
      return l;

  return (long long) a * b;
}

long long int gcd(long long int a, long long int b) {
  if(!b) 
    return a;
  else
  {
    return gcd(b, a%b);
  }
  
}

long long int lcm_fast(long long int a, long long int b) {
  long long int h = gcd(a, b);
  long long int l = a * (b / h);
  return l;
}

int main() {
  int a, b;
  std::cin >> a >> b;
  std::cout << lcm_fast(a, b) << std::endl;
  return 0;
}
